istioctl/pkg/rbac/upgrade_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rbac
+
+import (
+	"io/ioutil"
+	"reflect"
+	"testing"
+)
+
+const (
+	testFailedWithError   = "test failed with error %v"
+	testFailedExpectedGot = "test failed. Expected\n%sGot\n%s"
+)
+
+type testCases struct {
+	input    Upgrader
+	expected string
+}
+
+func TestUpgradeLocalFile(t *testing.T) {
+	cases := []testCases{
+		{
+			input: Upgrader{
```

Comment:
You don't need to embed the whole `Upgrader` here, instead, you can only put the input file `./testdata/rbac-policies.yaml` as a string, the mapping of `service-viewer` as a map and construct the `Upgrader` in line 52 or so.

Reply:
Discussed offline. Done